here-naksha-app-service/src/test/java/com/here/naksha/app/service/util/ApiUtilTest.java

```diff
@@ -18,102 +18,147 @@
+
     // validate 5th operation uses AND
-    crtOp = orList.get(i++);
-    assertEquals(POpType.AND, crtOp.op());
-    innerOpList = crtOp.children();
-    assertNotNull(innerOpList, "Expected multiple operations");
-    assertEquals(2, innerOpList.size());
-    assertEquals(POpType.EXISTS, innerOpList.get(0).op());
-    assertEquals("eight", innerOpList.get(0).getPropertyRef().getTagName());
-    assertEquals(POpType.EXISTS, innerOpList.get(1).op());
-    assertEquals("nine", innerOpList.get(1).getPropertyRef().getTagName());
+    assertThatOperation(orList.get(innerOpsInd++))
+        .hasType(OpType.AND)
+        .hasChildrenThat(eighth -> eighth.existsWithTagName("eight"), ninth -> ninth.existsWithTagName("nine"));
+
     // validate 6th operation uses AND
-    crtOp = orList.get(i++);
-    assertEquals(POpType.AND, crtOp.op());
-    innerOpList = crtOp.children();
-    assertNotNull(innerOpList, "Expected multiple operations");
-    assertEquals(2, innerOpList.size());
-    assertEquals(POpType.EXISTS, innerOpList.get(0).op());
-    assertEquals("ten", innerOpList.get(0).getPropertyRef().getTagName());
-    assertEquals(POpType.EXISTS, innerOpList.get(1).op());
-    assertEquals("eleven", innerOpList.get(1).getPropertyRef().getTagName());
+    assertThatOperation(orList.get(innerOpsInd++))
+        .hasType(OpType.AND)
+        .hasChildrenThat(
+            tenth -> tenth.existsWithTagName("ten"), eleventh -> eleventh.existsWithTagName("eleven"));
+
     // validate 7th operation uses OR
-    crtOp = orList.get(i++);
-    assertEquals(POpType.OR, crtOp.op());
-    innerOpList = crtOp.children();
-    assertNotNull(innerOpList, "Expected multiple operations");
-    assertEquals(2, innerOpList.size());
-    assertEquals(POpType.EXISTS, innerOpList.get(0).op());
-    assertEquals("twelve", innerOpList.get(0).getPropertyRef().getTagName());
-    assertEquals(POpType.EXISTS, innerOpList.get(1).op());
-    assertEquals("thirteen", innerOpList.get(1).getPropertyRef().getTagName());
+    assertThatOperation(orList.get(innerOpsInd++))
+        .hasType(OpType.OR)
+        .hasChildrenThat(
+            twelfth -> twelfth.existsWithTagName("twelve"),
+            thirteenth -> thirteenth.existsWithTagName("thirteen"));
+
     // validate 8th operation uses EXISTS
-    crtOp = orList.get(i++);
-    assertEquals(POpType.EXISTS, crtOp.op());
-    assertEquals("fourteen", crtOp.getPropertyRef().getTagName());
+    assertThatOperation(orList.get(innerOpsInd)).existsWithTagName("fourteen");
+  }
+
+  @ParameterizedTest
+  @MethodSource("simpleTagsSample")
+  void shouldParseSimpleTags(String queryString, Consumer<POpAssertion> assertion) {
+    QueryParameterList queryParameters = new QueryParameterList(queryString);
+    POp op = ApiUtil.buildOperationForTagsQueryParam(queryParameters);
+    assertion.accept(new POpAssertion(op));
+  }
+
+  private static Stream<Arguments> simpleTagsSample() {
```

Comment:
let's also validate the case that @gunplar suggested yesterday, i.e. when tags=,a+b .. expectation is that first empty tag should be omitted and final result should be AND condition between "a" and "b"

Reply:
It's way harder that it might seem at the first sight ;) Take a look at the `QueryParameterDecoder` class and its `parse` method - I've been analyzing it for a while and I think it would be best to not cover this functionality as part of this PR, I suggest creating a "nice to have" task for later to be picked up - the fix itself shouldn't be hard but analysis might consume some time that I belive will be better spent elsewhere (at least for now)